### PR TITLE
started on integration tests

### LINKED ISSUE #15: Add a basic integration test

### PR DESCRIPTION
**Pull request by erezmus**
Closes #15 

- Adds a very basic integration test to audit service
- Restructured set up code
- Use of MockEventBus
- Use of docker-compose.test.yml for postgres container

Todo:
- [x] run migrations for database
- [x] fix setup.test.ts
- [x] add test with sending events on message bus

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,6 +16,13 @@ lint: get_deps
 test: get_deps
 	yarn test
 
+test_integration:
+	- ${DOCKER_COMPOSE_TEST} down -v
```
Comment (Cooryd) [inline Makefile:20]: If we're stopping containers on L24 after this commands been run do we need to account for them being up on start?

File: src/app.test.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        beforeEach(() => {
+            (InfraLogger.info as jest.Mock).mockClear();
+            mockEventBus = ({
+                init: jest.fn().mockReturnThis(),
+                subscribe: jest.fn(),
+            } as unknown) as EventBus;
+        });
+
+        // for some reason uncommenting this gives timeout
+        // afterEach(async () => { await app.shutdown(); });
```
Comment (Cooryd) [inline src/app.test.ts]: If this isn't needed then, should we remove it from the test?

File: src/config.ts [hunk 3]
```diff
@@ -14,35 +14,36 @@ interface Config {
...
     port: number;
     knex: KnexConfig;
     event: EventConfig;
 };
 
-export const serviceConfig: ConfigType = {
-    port: Number(config.port) || 3004,
-    knex: knexConfig,
-    event: eventConfig,
+export const serviceConfig = (): ConfigType => {
```
Comment (Cooryd) [inline src/config.ts]: Should we rename this to be `getServiceConfig` to make it more implicit it needs to be invoked?

File: src/main.ts [hunk 4]
```diff
@@ -1,10 +1,16 @@
...
-
-setup(config).then(app =>
-    app.listen(config.port, () => {
-        logger.info(`Audit service listening on port ${config.port}`);
-    }),
-);
+import { serviceConfig } from './config';
+import App from './app';
+
+const main = async (): Promise<void> => {
```
Comment (Cooryd) [inline src/main.ts:7]: Can we export this function and add a unit test to ensure its calling the functions expected and passing the return of `serviceConfig` to the `App` constructor.

Comment (Cooryd) [reply src/main.ts:7]: This was why we had the startup code extracted to the `setup.ts` file as the execution on import is a pain for unit testing.

File: tests/login.test.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+                timestamp: date,
+            },
+        };
+
+        await eventBus.publish(event);
+
+        const knex = app.getKnex();
+        const result = await knex.table('audit').whereExists(knex.select('*').from('audit').whereRaw('audit.entity = \'user:user_Id\''));
+
+        expect(result).toHaveLength(1);
```
Comment (Cooryd) [inline tests/login.test.ts:41]: Should we be asserting all of the data emitted is ending up in the DB correctly?

File: Makefile (hunk 1 above)
Comment (erezmus) [reply Makefile:20]: i think this is more for local development. sometimes a test fails or you are in an inconsistent state with the containers so this makes sure it starts fresh.

**Review by Cooryd — APPROVED** [review]

**Pull request merged**